File: packages/sdk/typescript/human-protocol-sdk/src/graphql/queries/kvstore.ts
import gql from 'graphql-tag';

const KVSTORE_FRAGMENT = gql`
  fragment KVStoreFields on KVStore {
    id
    block
    timestamp
    address
    key
    value
  }
`;

export const GET_KVSTORE_BY_ADDRESS_QUERY = () => {
  return gql`
    query getKVStoreData($address: String!) {
      kvstores(where: { address: $address }) {
        ...KVStoreFields
      }
    }
    ${KVSTORE_FRAGMENT}
  `;
};

export const GET_KVSTORE_BY_ADDRESS_AND_KEY_QUERY = () => {
  return gql`
    query getKVStoreDataByKey($address: String!, $key: String!) {
      kvstores(where: { address: $address, key: $key }) {
        ...KVStoreFields
      }
    }
    ${KVSTORE_FRAGMENT}
  `;
};

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/gql/kvstore.py
kvstore_fragment = """
fragment KVStoreFields on KVStore {
    id
    block
    timestamp
    address
    key
    value
}
"""


def get_kvstore_by_address_query() -> str:
    return f"""
query getKVStoreData($address: String!) {{
    kvstores(
        where: {{
            address: $address
        }}
    ) {{
      ...KVStoreFields
    }}
}}
{kvstore_fragment}
"""


def get_kvstore_by_address_and_key_query() -> str:
    return f"""
query getKVStoreDataByKey($address: String!, $key: String!) {{
    kvstores(
        where: {{
            address: $address,
            key: $key
        }}
    ) {{
      ...KVStoreFields
    }}
}}
{kvstore_fragment}
"""
